src/smt/theory_str_noodler/regex.cpp

```diff
@@ -0,0 +1,559 @@
+                res.universal = l_false;
+                res.empty = l_false;
+            } else if(res.universal == l_true) {
+                res.universal = l_false;
+                res.empty = l_true;
+            } else {
+                res.universal = l_undef;
+                res.empty = l_undef;
+            }
+            return res;
+        } else if (m_util_s.re.is_derivative(expression)) { // Handle derivative.
+            util::throw_error("derivative is unsupported");
+        } else if (m_util_s.re.is_diff(expression)) { // Handle diff.
+            util::throw_error("regex difference is unsupported");
+        } else if (m_util_s.re.is_dot_plus(expression)) { // Handle dot plus.
+            return RegexInfo{.min_length = 1, .universal = l_false, .empty = l_false};
+        } else if (m_util_s.re.is_empty(expression)) { // Handle empty language.
+            return RegexInfo{.min_length = 0, .universal = l_false, .empty = l_true};
+        } else if (m_util_s.re.is_epsilon(expression)) { // Handle epsilon.
+            return RegexInfo{.min_length = 0, .universal = l_false, .empty = l_false};
+        } else if (m_util_s.re.is_full_char(expression)) { // Handle full char (single occurrence of any string symbol, '.').
+            return RegexInfo{.min_length = 1, .universal = l_false, .empty = l_false};
+        } else if (m_util_s.re.is_full_seq(expression)) {
+            return RegexInfo{.min_length = 0, .universal = l_true, .empty = l_false};
+        } else if (m_util_s.re.is_intersection(expression)) { // Handle intersection.
+            SASSERT(expression->get_num_args() > 0);
+            // min_length: maximum of each regex from intersection
+            // empty: undef
+            // universal: min_length > 0 --> false; otherwise undef
+            RegexInfo res = get_regex_info(to_app(expression->get_arg(0)), m_util_s, m);
+            for (unsigned int i = 1; i < expression->get_num_args(); ++i) {
+                RegexInfo prod = get_regex_info(to_app(expression->get_arg(i)), m_util_s, m);
+                res.min_length = std::max(res.min_length, prod.min_length);
+            }
+            res.empty = l_undef;
+            res.universal = l_undef;
+            if(res.min_length > 0) {
+                res.universal = l_false;
+            }
+            return res;
+        } else if (m_util_s.re.is_loop(expression)) { // Handle loop.
+            unsigned low, high;
+            expr *body;
+            if (m_util_s.re.is_loop(expression, body, low, high)) {
+            } else if (m_util_s.re.is_loop(expression, body, low)) {
+            } else {
+                util::throw_error("loop should contain at least lower bound");
+            }
+
+            // min_length: low == 0 --> 0; otherwise min_length * low
+            // empty: low == 0 --> false; otherwise the same as the original empty
+            // universal: min_length --> false; low == 0 --> false
```

Comment:
Should it be
```suggestion
            // universal: min_length > 0 --> false; empty && low == 0 --> false
```
?
Also I think if `low > 0`, then universal should be set to false.